rspace/src/main/scala/coop/rchain/rspace/HotStore.scala

```diff
@@ -256,112 +261,118 @@ private class InMemHotStore[F[_]: Concurrent, C, P, A, K](
     val (l1, l2) = col splitAt index
-    (l1 ++ (l2 tail))
+    l1 ++ (l2 tail)
   }
 
   def toMap: F[Map[Seq[C], Row[P, A, K]]] =
     for {
       cache         <- hotStoreState.get
-      data          = cache.data.map(_.leftMap(Seq(_))).toMap
-      continuations = (cache.continuations ++ cache.installedContinuations.mapValues(Seq(_))).toMap
-      zipped        = zip(data, continuations, Seq.empty[Datum[A]], Seq.empty[WaitingContinuation[P, K]])
-      mapped        = zipped.mapValues { case (d, k) => Row(d, k) }
+      data          = cache.data.map(_.leftMap(Seq(_)))
+      continuations = cache.continuations ++ cache.installedContinuations.mapValues(Seq(_))
+
+      dataAt = (i: Seq[C]) => data.getOrElse(i, Seq.empty[Datum[A]])
+      continuationAt = (i: Seq[C]) =>
+        continuations.getOrElse(i, Seq.empty[WaitingContinuation[P, K]])
+      mapped = (data.keySet ++ continuations.keySet).map { i =>
+        (i, Row(dataAt(i), continuationAt(i)))
+      }.toMap
     } yield mapped.filter { case (_, v) => !(v.data.isEmpty && v.wks.isEmpty) }
 
   private def getContFromHistoryStore(channels: Seq[C]) =
     for {
       d <- Deferred[F, Seq[WaitingContinuation[P, K]]]
-      r <- historyStoreCache
-            .modify[(Deferred[F, Seq[WaitingContinuation[P, K]]], Boolean)](cache => {
-              cache.continuations.get(channels) match {
-                case Some(v) => (cache, (v, true))
-                case None =>
-                  (cache.copy(continuations = cache.continuations + (channels -> d)), (d, false))
-              }
-            })
+      r <- historyStoreCache.modify { cache =>
+            cache.continuations
+              .get(channels)
+              .map(v => (cache, (v, true)))
+              .getOrElse(
+                cache.copy(continuations = cache.continuations + (channels -> d)),
+                (d, false)
+              )
+          }
       (deferred, isComplete) = r
-      _                      <- (historyReaderBase.getContinuations(channels) >>= deferred.complete).whenA(!isComplete)
+      continuations          <- historyReaderBase.getContinuations(channels)
+      _                      <- deferred.complete(continuations) whenA !isComplete
       contInHistoryStore     <- deferred.get
     } yield contInHistoryStore
 
   private def getDataFromHistoryStore(channel: C) =
     for {
       d <- Deferred[F, Seq[Datum[A]]]
-      r <- historyStoreCache
-            .modify[(Deferred[F, Seq[Datum[A]]], Boolean)](cache => {
-              cache.datums.get(channel) match {
-                case Some(v) => (cache, (v, true))
-                case None    => (cache.copy(datums = cache.datums + (channel -> d)), (d, false))
-              }
-            })
+      r <- historyStoreCache.modify { cache =>
+            cache.datums
+              .get(channel)
+              .map(v => (cache, (v, true)))
+              .getOrElse(cache.copy(datums = cache.datums + (channel -> d)), (d, false))
```

Comment:
Please change tuple definition with `->` as other tuples. Both are ok but mixing it in the same function looks confusing.

Reply:
@tgrospic, should it also be replaced in HotStoreSpec?

Reply:
It's fine.